src/models/state/wallet/sent_transaction.rs

```diff
@@ -0,0 +1,42 @@
+use serde::Deserialize;
+use serde::Serialize;
+use tasm_lib::prelude::Digest;
+
+use super::unlocked_utxo::UnlockedUtxo;
+use crate::models::blockchain::type_scripts::native_currency_amount::NativeCurrencyAmount;
+use crate::models::proof_abstractions::timestamp::Timestamp;
+use crate::models::state::transaction_details::TransactionDetails;
+use crate::models::state::wallet::transaction_output::TxOutputList;
+
+/// represents a user-level tx that has been sent by this wallet.
+///
+/// this type is intended for storing in the wallet-db in order to
+/// group together inputs and outputs as a single payment for purposes
+/// of wallet history display.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SentTransaction {
+    pub tx_inputs: Vec<UnlockedUtxo>,
```

Comment:
`UnlockedUtxo` contains secret key material. So with this change, the database will contain secret keys. (Unless you're guessing) that wasn't the case earlier. Is that a good idea? Can we avoid the secret keys that are contained in the `LockScriptAndWitness`  in these database entries?

Reply:
addressed in 3048aa3ff69bdf84d601a544861f8a40e54078fe